### PR TITLE
Add `relatedResource` and `digestSRI` to the vocabulary

### PR DESCRIPTION
**Pull request by iherman**
This is the [action I took at the F2F](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2023-09-14-vcwg#section5-1): adding the resource integrity terms to the VCDM vocabulary. Here is what I did:

- Added anchors to the `digestSRI` and `relatedResource` terms in the specification itself (to be used by the vocabulary)
- Added a section to define the `sriString` datatype
- Added the `digestSRI`, `relatedResource`, `mediaType` to the v2 `@context` file
- Added the `digestSRI` and `relatedResource` properties, as well as the `sriString` datatype, to `vocabulary.yml`
- Added a section on data types in the vocabulary’s `template.html` file for the vocabulary generation
- Expanded the diagram’s alt text to also refer to the new terms

Note that the `mediaType` term has not been added to the vocabulary; instead, it is mapped to https://schema.org/encodingFormat term. Let us avoid the "not invented here" syndrome...

Not strictly related to the underlying issue, but:

- Added the `vocabulary.drawio` file to the repository and expanded the `README.md` file in the corresponding directory (this makes the maintenance of the diagrams easier and puts the raw data into our repo)
- Updated the diagram (there were syntax errors in the links)

The automatic preview below works for the spec text, but not for the vocabulary. For the vocabulary files, see the [separate vocabulary preview](https://w3c.github.io/yml2vocab/previews/vcdm/)

This PR takes care of the issues #1237 and #1265.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/vc-data-model/pull/1296.html" title="Last updated on Oct 16, 2023, 7:24 AM UTC (b4031ef)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-data-model/1296/286c2b2...b4031ef.html" title="Last updated on Oct 16, 2023, 7:24 AM UTC (b4031ef)">Diff</a>

### CODE REVIEW

**Comment by iherman** [comment]
Note: the SRI spec defines the [integrity attribute](https://www.w3.org/TR/SRI/#the-integrity-attribute) in such a way that the attribute value can be a _list_ of hash expressions (see [Example 7](https://www.w3.org/TR/SRI/#does-response-match-metadatalist) in the spec). By formally referring to those sections, the VCDM inherits this feature. I do not think that is a problem, I would rather think this _is_ a feature, but I just wanted to draw attention to this fact.

**Comment by iherman** [comment]
To be a bit picky about the RDFS aspects of the vocabulary: at the moment, the range of the `relatedResource` (resp.  the domain of `digestSRI`) is unspecified, except to say that the domain of `relatedResource` must be an object with a URI (i.e., not a literal). I was tempted to use https://schema.org/CreativeWork for both, which is a class for very general, but nevertheless “concrete” objects (as opposed to abstract concepts like “the Moon”, which can also have a URI in the RDF world). Using `CreativeWork` makes sense because there is no way to calculate the hash of the Moon… That assignment would also be in line with the fact that the `mediaType` property has been aliased (in the context file) to https://schema.org/encodingFormat, whose domain is indeed `CreativeWork`. However, I did not want to use this class without a further discussion with, and consensus of, the Working Group.

I am happy to update the files if there is a consensus on this detail.

Cc @pchampin

**Review by decentralgabe — APPROVED** [review]

File: index.html
Comment (TallTed) [inline index.html:6216]:
```suggestion
                The canonical mapping consists of the lexical-to-value mapping.
```

File: vocab/credentials/v2/README.md
Comment (TallTed) [inline vocab/credentials/v2/README.md:16]:
```suggestion
- `template.json`: an HTML template file used by the script; it is the skeleton of the final HTML format based on [ReSpec](https://respec.org/docs/). If the file is modified, care should be taken not to change the core structure and the various, possibly empty, HTML elements with `@id` values. The script fills those elements with content when generating the `vocabulary.html` file (and removes any sections that remain empty after processing).
```

**Review by TallTed — APPROVED** [review]

**Review by msporny — APPROVED** [review]

**Review by seabass-labrax — APPROVED** [review]
I would also be happy if this used https://schema.org/CreativeWork.

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2023-10-11-vcwg#section1-9) on 2023-10-11

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 1.9. Add `relatedResource` and `digestSRI` to the vocabulary (pr vc-data-model#1296)

_See github pull request [vc-data-model#1296](https://github.com/w3c/vc-data-model/pull/1296)._




**Brent Zundel:** One approval, a bit of explanation from Ivan.  
… This feels like a PR that just should be merged ... to me, but happy to hear otherwise.  
… I'm not hearing otherwise, encourage folks to jump on, review and approve to get it in.  

> *Manu Sporny:* +1 to this PR.

</details>

File: vocab/credentials/v2/vocabulary.drawio
Comment (davidlehn) [inline vocab/credentials/v2/vocabulary.drawio:7]: All the links in here are w3.org vs www.w3.org and with a trailing `/`.  The server redirs to www and adds the `/` no matter what you do, but the spec and context say vocab is `https://www.w3.org/2018/credentials#` (https://www.w3.org/TR/vc-data-model-2.0/#vocabularies).  Should these links match the vocab URLs?

File: vocab/credentials/v2/vocabulary.yml
Comment (davidlehn) [inline vocab/credentials/v2/vocabulary.yml:168]: Copy & paste typo.  I think this is the correct link:
```suggestion
    defined_by: https://www.w3.org/TR/vc-data-model-2.0/#defn-relatedResource
```

**Review by davidlehn — COMMENTED** [review]
- Attempted to align `sriString` capitalization and URL.
- Comments on vocab URL confusion.  It seems like the vocab doc, spec, and context should be using strictly the same URL strings for identifiers.  Or am I missing something there?
- Were the diagram object fill colors an accessibility problem?

File: vocab/credentials/v2/vocabulary.yml
Comment (iherman) [reply vocab/credentials/v2/vocabulary.yml:166]: This is correct, and I will change this. Note that it is an inherited bug: it seems that the `w3.org` URL was used in earlier versions, so we should make sure that the change is done everywhere. 

cc @msporny, @dlongley

File: contexts/credentials/v2
Comment (msporny) [inline contexts/credentials/v2:76]: Hmm, do we want both of these as top-level values... or do we want them under `relatedResource`. I would imagine we'd want the latter?

**Comment by msporny** [comment]
Waiting on updates from @iherman to merge.

File: vocab/credentials/v2/vocabulary.yml
Comment (iherman) [reply vocab/credentials/v2/vocabulary.yml:166]: I have done the change in https://github.com/w3c/vc-data-model/pull/1296/commits/16eaf39fc672dc661f17a042f8fbe0dfdf14f72b, marking this as resolved.

File: vocab/credentials/v2/vocabulary.drawio
Comment (iherman) [reply vocab/credentials/v2/vocabulary.drawio:7]: I have hit some problems with dereferencing these links properly. For some reason, the dereferencing from the embedded version did not work properly without the extra '/'.

@davidlehn, is it o.k. if we raise this as a separate editorial issue marked as post-CR? I would not want to keep this PR on hold (this PR is labeled as before CR)

Cc @msporny

File: contexts/credentials/v2
Comment (iherman) [reply contexts/credentials/v2:76]: @msporny I am open either way, I leave the decision to you guys.

My argument for this setup is that `digestSRI` might be useful elsewhere; there is nothing in the definition that is really bound to `relatedResource`. After all, this is a general mechanism. (Note that the domain of `digestSRI` is essentially open-ended, see also https://github.com/w3c/vc-data-model/pull/1296#issuecomment-1742010597). But, again, I regard this as a VCDM domain issue, and I am not a VCDM expert, so to say...

File: vocab/credentials/v2/vocabulary.drawio
Comment (iherman) [reply vocab/credentials/v2/vocabulary.drawio:487]: @davidlehn, this has been covered in https://github.com/w3c/vc-data-model/pull/1296/commits/512faeb4abc07ec6491559f4f77c420645665f31. Mark this as resolved.

**Comment by iherman** [comment]
> Waiting on updates from @iherman to merge.

@msporny made the changes, see https://github.com/w3c/vc-data-model/pull/1296#discussion_r1360172050 and https://github.com/w3c/vc-data-model/pull/1296#discussion_r1360176772. I have settled everything else.

For the former: I will look at this issue again, but I would not want the PR to be held on this one. I promise to look into the issue in any case, just do not know when.

For the latter: I leave you the decision. It is fine if you guys do the change on the context file if you decide to do so and then merge the PR.

File: vocab/credentials/v2/vocabulary.drawio
Comment (iherman) [reply vocab/credentials/v2/vocabulary.drawio]: Scrap my previous remark. I have solved the issue (in https://github.com/w3c/vc-data-model/pull/1296/commits/b4031ef5d9c8fab6084b098730951992477bdb62), and refreshed the files accordingly. Resolving this conversation as well.

Thanks for these, @davidlehn !

**Comment by msporny** [comment]
I'm going to merge this, pending the discussion on `mediaType` and `digestSRI` -- while the latter is useful everywhere, the former might conflict w/ other vocabularies. Whatever changes we'd need to make would be in the context, so we can probably just merge this for now and make changes later (if necessary).

**Pull request merged**
